controller-server/src/main/java/com/yahoo/vespa/hosted/controller/role/RoleMembership.java

```diff
@@ -52,4 +58,59 @@ public String toString() {
         RoleMembership membership(Principal user);
     }
 
+    public interface Builder {
+
+        BuilderWithRole add(Role role);
+
+        RoleMembership build();
+
+    }
+
+    public static class BuilderWithRole implements Builder {
+
+        private final SystemName system;
+        private final Map<Role, Set<Context>> roles;
+
+        private Role current;
+
+        private BuilderWithRole(SystemName system) {
+            this.system = Objects.requireNonNull(system);
+            this.roles = new HashMap<>();
+        }
+
+        @Override
+        public BuilderWithRole add(Role role) {
+            if (current != null) {
+                roles.putIfAbsent(current, new HashSet<>());
+                roles.get(current).add(Context.unlimitedIn(system));
+            }
+            current = role;
+            return this;
+        }
+
+        public Builder limitedTo(TenantName tenant) {
+            roles.putIfAbsent(current, new HashSet<>());
+            roles.get(current).add(Context.limitedTo(tenant, system));
+            current = null;
+            return this;
+        }
+
+        public Builder limitedTo(TenantName tenant, ApplicationName application) {
+            roles.putIfAbsent(current, new HashSet<>());
+            roles.get(current).add(Context.limitedTo(tenant, application, system));
```

Comment:
Consider extracting a method for these two lines that takes context as argument, to remove the repetition.

Reply:
Sure.